### PR TITLE
[REST API] Display the Jetpack benefits banner and modal for Application Passwords users

### LINKED ISSUE #8365: [REST API] Benefits banner and modal
Reuse the Jetpack CP benefits banner and modal as an entry point to the Jetpack installation:

- [x] Display the banner when application passwords are used.
- [x] Update the modal's content depending on the site's type.
- [x] Before starting the flow, fetch the Jetpack API to detect the status of the Jetpack.
- [x] Display the "install Jetpack" button in settings

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #8365 

### Description
Internal docs: pe5sF9-19y-p2

As explained in the issue, these are the tasks that this PR does:
- Display the banner when application passwords are used.
- Update the modal's content depending on the site's type.
- Fetch the Jetpack API to detect the status of the Jetpack before attempting the authentication and installation.
- Display the "install Jetpack" button in the settings

The other main change is: adding a feature flag for the project. 

And lastly, although it wasn't a fundamental necessity, I migrated the screen to Compose, this just made handling the different states and the loading dialog easier.

**Note**: the number of lines is misleading, it's just because I moved the files to a different package to group them.

### Testing instructions
1. Open the app, and click on "enter a site address".
2. Enter the address of a non-Jetpack site.
3. Enter your site's credentials.
4. In the dashboard screen, confirm that you can see the Jetpack benefits banner.
5. Click on the banner.
6. Check the content of the modal (the icon for the third point needs updating, I'll update it in a separate PR when we get it from the design team).
7. Click on "Log in to continue"
8. Confirm a loading dialog is shown, then a toast is shown.
9. Go back to the dashboard.
10. Open the settings screen.
11. Confirm that the "Install Jetpack" button is shown.
12. Click on the button, and confirm it opens the Jetpack benefits modal.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ParaskP7** [comment]
FYI: @hichamboushaba @JorgeMucientes @0nko I moved the `12.5` milestone to `12.6` in order to proceed with the code freezing process. Do let me know if you still need this PR to be part of the `12.5` release.

**Comment by JorgeMucientes** [comment]
Hey @ParaskP7, Hicham is AFK today. But I think we don't need to block the code freeze because of this PR, the feature is afaik featured flagged and WIP, meaning it won't be available for users. 

In any case I'll proceed to review this now.

**Comment by ParaskP7** [comment]
> Hey @ParaskP7, Hicham is AFK today. But I think we don't need to block the code freeze because of this PR. In any case I'll proceed to review this now.

Thanks so much for the update @JorgeMucientes , much appreciated! 🙇 

FYI: No matter, we could always target the `release/12.5` branch afterwards and get that there if need be. 💯

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt [hunk 1]
```diff
@@ -343,8 +344,9 @@ class MyStoreViewModel @Inject constructor(
             System.currentTimeMillis() - appPrefsWrapper.getJetpackBenefitsDismissalDate()
         )
         // For now, the banner is shown only when the site is a Jetpack CP site
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: Super minor np, is this comment really needed? I think the code is clear and self explanatory

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Excellent job @hichamboushaba, code looks great and everything works as expected. I left a super minor np, but feel free to ignore it. 

Ready to 🚢 

> FYI: No matter, we could always target the release/12.5 branch afterwards and get that there if need be. 💯


Yes, that is true @ParaskP7 👍🏼 , but tbh I prefer to be strict on what we merge into `release-vX.X` branches. Imho only bugs or crash fixes should be merged there and not features that didn't make it to the freeze. Unless there's a very good reason for it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: That's actually a nice catch, this comment was there before and I should've removed it, as the banner will be shown for both types now, I'll remove it right now.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: Done in [408cb7b](https://github.com/woocommerce/woocommerce-android/pull/8374/commits/408cb7b92e65877f4296b615599482b19c53938a)

**Comment by ParaskP7** [comment]
> Yes, that is true @ParaskP7 👍🏼 , but tbh I prefer to be strict on what we merge into release-vX.X branches. Imho only bugs or crash fixes should be merged there and not features that didn't make it to the freeze.

💯 

> Unless there's a very good reason for it.

Exactly, this is what I meant! 🥇

**Pull request merged**